go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -1532,33 +1542,124 @@ func (m *valueMerger) tryMerge(left, right, base val.Tuple) (val.Tuple, bool) {
 
 // processColumn returns the merged value of column |i| of the merged schema,
 // based on the |left|, |right|, and |base| schema.
-func (m *valueMerger) processColumn(i int, left, right, base val.Tuple) ([]byte, bool) {
+func (m *valueMerger) processColumn(ctx context.Context, i int, left, right, base val.Tuple) ([]byte, bool) {
 	// missing columns are coerced into NULL column values
 	var leftCol []byte
-	if l := m.leftMapping[i]; l != -1 {
-		leftCol = left.GetField(l)
+	leftColumnIndex := m.leftMapping[i]
+	if leftColumnIndex != -1 {
+		leftCol = left.GetField(leftColumnIndex)
 	}
 	var rightCol []byte
+	rightColumnIndex := m.rightMapping[i]
 	if r := m.rightMapping[i]; r != -1 {
 		rightCol = right.GetField(r)
 	}
 
-	if m.vD.Comparator().CompareValues(i, leftCol, rightCol, m.vD.Types[i]) == 0 {
-		return leftCol, false
+	if base == nil || m.baseMapping[i] == -1 {
+		// There are three possible cases:
+		// - The base row doesn't exist, or
+		// - The column doesn't exist in the base row, or
+		// - The column was default-null and not reflected in the tuple. (TODO: verify this is handled correctly)
+		// Regardless, either both left and right are inserts, or one is an insert and the other
+		// is a no-op. If they're inserts of different types, then we would have already detected a conflict.
+		// Thus, we can assume that there is no schema change conflict here.
+		if m.resultVD.Comparator().CompareValues(i, leftCol, rightCol, m.resultVD.Types[i]) == 0 {
+			// columns are equal, return either.
+			return leftCol, false
+		}
+		leftModified := leftCol != nil
+		rightModified := rightCol != nil
+
+		switch {
+		case leftModified && rightModified:
+			// conflicting inserts
+			return nil, true
+		case leftModified:
+			return leftCol, false
+		default:
+			return rightCol, false
```

Comment:
So, if a row with the same PK was inserted on both sides of the merge, then we look at each column and if one column is nil, then we always select the other column? I'm not sure if that's the right thing to do... it seems like it should just be a conflict that users have to resolve, since we don't know if the value or nil is the right choice to keep. Let me know if I'm just missing something. 😄

Reply:
Yeah, that wasn't correct. The new implementation should handle that better.